includes/Modules/Reader_Revenue_Manager.php

```diff
@@ -458,14 +487,43 @@ public function register_tag() {
 			return;
 		}
 
+		$post_product_id = '';
+
+		if (
+			Feature_Flags::enabled( 'rrmModuleV2' ) &&
+			is_singular()
+		) {
+			$post_product_id = $this->post_product_id->get( get_the_ID() );
+		}
+
 		$tag->use_guard( new Tag_Verify_Guard( $this->context->input() ) );
-		$tag->use_guard( new Tag_Guard( $module_settings ) );
+		$tag->use_guard( new Tag_Guard( $module_settings, $post_product_id ) );
```

Comment:
It's worth noting that `$post_product_id` will also be in the format `{publicationID}:{productID}` at this point if it's set to a product ID. However, it looks like we don't need to extract it because we only need to check for `none` or it otherwise being empty in the tag guard.

So, nothing to do here, just making a note for future reference.

Reply:
Exactly, thanks for summarising!